Build baw client

Allcharles opened the pull request:
# Build Baw Client

Build baw client in github actions, this allows the unit tests to validate that the baw client has been correctly integrated into the project.

## Changes

- Added tests for baw client pages
- Updated actions workflow to build baw client
- Added resolvers to baw client pages so that if the model is not available, we use the projects error page instead of the baw client error handler

## Final Checklist

- [ ] Assign reviewers if you have permission
- [ ] Assign labels if you have permission
- [ ] Link issues related to PR
- [ ] Ensure project linter is not producing any warnings (`npm run lint`)
- [ ] Ensure build is passing on all browsers (`npm run test:all`)
- [ ] Ensure CI build is passing
- [ ] Ensure docker container is passing ([docs](https://github.com/QutEcoacoustics/workbench-client#docker))

Allcharles committed:
Build Baw Client (#839)

Build baw client in github actions, this allows the unit tests to validate that the baw client has been correctly integrated into the project.

atruskie reviewed (approved):
I'd suggest leaving a common comment tag to any place this change affects so we can rip it out later.

```
// TODO: OLD-CLIENT REMOVE
```

Allcharles committed:
Working Login (#853)

* Cleanup app styling to create full height components

- Cleanup CSS styling for the app component so that the default router outlet component will have access to the full height of the page
- Created basic audio analysis page

* Working example

Couple bugs to fix
- AngularJS links are not reloading the parent page
- IFrame does not resize to content which causes the scrollbar to be in the wrong location

* Created new audio analysis page

* Cleanup iframe module

- Disabled broken tests
- Renamed IFrameComponent to BawClientComponent
- Isolated BawClientComponent into a module
- Created placeholder unit tests
- Updated interceptor to include cookies in requests

* Implemented BawClientComponent unit tests

* Fixed broken tests and rebase conflicts

* Fixed broken unit tests

Discovered bug in how secondary menu unit tests were saving/restoring the defaultMenu global variable (converted Lists to arrays, but not back)

* Fix site/region details tag links

* Working login request

* Update security service tests

* Cleanup interceptor tests

* Fix rebase issues

* Cleanup api request responseType's

This should make things a lot more clear for when the interceptor should set accept/content-type headers to requests

* Cleanup review comments
